interfaces/policy/helpers.go

```diff
@@ -44,6 +45,27 @@ func checkSnapType(snapType snap.Type, types []string) error {
 	return fmt.Errorf("snap type does not match")
 }
 
+func checkID(kind, id string, ids []string, special map[string]string) error {
+	if len(ids) == 0 {
+		return nil
+	}
+	if id == "" { // unset values never match
+		return fmt.Errorf("%s does not match", kind)
+	}
```

Comment:
Yes, unasserted. Per irc, this will cause unasserted snaps with no snap id or publisher id to fail when there is a constraint on either of these, which is correct behavior.